data/models/plant.py

```diff
@@ -53,16 +53,34 @@ class Meta:
         verbose_name="famille de plante",
         related_name="ca_plant_set",
     )
-    # TODO: output_field should be a ForeignKey
-    family = models.GeneratedField(
+    # TODO: ce champ n'est pas utile en tant que tel, il serait possible de l'éviter en créant un Field custom ForeignGeneratedField(ForeigObject)
+    family_by_id = models.GeneratedField(
         expression=Coalesce(F("ca_family"), F("siccrf_family")),
         output_field=models.BigIntegerField(verbose_name="famille de plante"),
         db_persist=True,
     )
+    family = models.ForeignObject(
```

Comment:
Vu que `ForeignObject` sera enlevé en Django 6.0 ([doc](https://docs.djangoproject.com/en/5.0/internals/deprecation/#deprecation-removed-in-6-0)), pourquoi pas une ForeignKey ?

Reply:
Je connaissais pas du tout ! Curieux du coup :)

je crois que c'est seulement 2 méthodes spécifiques qui vont être retirées dans la v6.0 nan ?
<img width="529" alt="image" src="https://github.com/betagouv/complements-alimentaires/assets/1933516/c9851edd-add2-4da9-a991-d5e6b55b567d">

Reply:
Oui, comme @ddahan c'est seulement 2 méthodes qui vont être dépréciées (et remplacées par d'autres) : https://github.com/django/django/blob/53719d6b5b745dd99b1ab9315afb242f706ebbf1/django/db/models/fields/related.py#L776
C'est plutôt une attention à avoir en cas d'héritage de ForeignObject.
Pas de souci dans ce cas à priori.
Je merge.